bin/metaBackend.py

```diff
@@ -21,47 +21,506 @@
+            schemaVersion = "unknown"
+            schemaDescr = ""
+        else:
+            (schemaVersion, schemaDescr) = ret
+
+        # This can be sometimes handy for debugging
+        # pp = pprint.PrettyPrinter(indent=2)
+        # pp.pprint(theTable)
+
+        # Get host/port from authFile
+        dict = readCredentialFile(dbMysqlAuthF, self._log)
+        (host, port) = [dict[k] for k in ('host', 'port')]
+
+        # Now, we will be talking to the metaserv database, so change
+        # connection as needed
+        if self._msMysqlAuthF != dbMysqlAuthF:
+            db = Db(read_default_file=self._msMysqlAuthF)
+
+        # get ownerId, this serves as validation that this is a valid owner name
+        ret = db.execCommand1("SELECT userId FROM User WHERE mysqlUserName = %s",
+                              (owner,))
+        if not ret:
+            self._log.error("Owner '%s' not found.", owner)
+            raise MetaBException(MetaBException.OWNER_NOT_FOUND, owner)
+        ownerId = ret[0]
+
+        # get projectId, this serves as validation that this is a valid project name
+        ret = db.execCommand1("SELECT projectId FROM Project WHERE projectName =%s",
+                              (projectName,))
+        if not ret:
+            self._log.error("Project '%s' not found.", owner)
+            raise MetaBException(MetaBException.PROJECT_NOT_FOUND, projectName)
+        projectId = ret[0]
+
+        # Finally, save things in the MetaServ database
+        cmd = "INSERT INTO Repo(url, projectId, repoType, lsstLevel, dataRelease, "
+        cmd += "version, shortName, description, ownerId) "
+        cmd += "VALUES('/dummy',%s,'db',%s,%s,%s,%s,%s,%s) "
+        opts = (str(projectId), level, dataRel, schemaVersion, dbName, schemaDescr,
+                str(ownerId))
+        db.execCommand0(cmd, opts)
+        repoId = db.execCommand1("SELECT LAST_INSERT_ID()")[0]
+        cmd = "INSERT INTO DbMeta(dbMetaId, dbName, connHost, connPort) "
+        cmd += "VALUES(%s,%s,%s,%s)"
+        db.execCommand0(cmd, (str(repoId), dbName, host, str(port)))
+
+        for t in theTable:
+            cmd = 'INSERT INTO DDT_Table(dbMetaId, tableName, descr) '
+            cmd += 'VALUES(%s, %s, %s)'
+            db.execCommand0(cmd, (str(repoId), t,
+                                  theTable[t].get("description", "")))
+            tableId = db.execCommand1("SELECT LAST_INSERT_ID()")[0]
+            isFirst = True
+            for c in theTable[t]["columns"]:
+                if isFirst:
+                    cmd = 'INSERT INTO DDT_Column(columnName, tableId, '
+                    cmd += 'ordinalPosition, descr, ucd, units) VALUES '
+                    opts = ()
+                    isFirst = False
+                else:
+                    cmd += ', '
+                cmd += '(%s, %s, %s, %s, %s, %s)'
+                opts += (c["name"], str(tableId), str(c["ord_pos"]),
```

Comment:
Same here, no str(). This multi-insert can be simplified with `cursor.executemany()` (if it was exposed in db API).